frontend/lib/src/components/shared/ElementFullscreen/ElementFullscreenWrapper.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {
+  FC,
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react"
+
+import { useTheme } from "@emotion/react"
+
+import { StyledFullScreenFrame } from "@streamlit/lib/src/components/shared/FullScreenWrapper/styled-components"
+import { ElementFullscreenContext } from "@streamlit/lib/src/components/shared/ElementFullscreen/ElementFullscreenContext"
+import { WindowDimensionsContext } from "@streamlit/lib/src/components/shared/WindowDimensions"
+import { useRequiredContext } from "@streamlit/lib/src/hooks/useRequiredContext"
+import { EmotionTheme } from "@streamlit/lib/src/theme"
+
+type ElementFullscreenWrapperProps = PropsWithChildren<{
+  height?: number
+  width: number
+}>
+
+const ElementFullscreenWrapper: FC<ElementFullscreenWrapperProps> = ({
+  children,
+  height,
+  width,
+}) => {
+  const theme: EmotionTheme = useTheme()
+  const [expanded, setExpanded] = useState(false)
+  const { fullHeight, fullWidth } = useRequiredContext(WindowDimensionsContext)
+
+  const zoomIn = useCallback(() => {
+    document.body.style.overflow = "hidden"
+    setExpanded(true)
+  }, [])
+
+  const zoomOut = useCallback(() => {
+    document.body.style.overflow = "unset"
+    setExpanded(false)
+  }, [])
+
+  const controlKeys = useCallback(
+    (event: KeyboardEvent) => {
+      // Your logic for handling keydown events
+      if (event.keyCode === 27 && expanded) {
```

Comment:
nit: comment for those of us who don't have the ascii table memorized. not me of course 😅😨

Reply:
Yeah, this was pretty much copy/pasted from https://github.com/streamlit/streamlit/blob/develop/frontend/lib/src/components/shared/FullScreenWrapper/FullScreenWrapper.tsx#L89 but agree, a comment is nice!